sdk_extensions/resources/src/main/java/io/opentelemetry/sdk/extensions/resources/ProcessResource.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.resources;
+
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import io.opentelemetry.sdk.resources.ResourceProvider;
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+
+/** {@link ResourceProvider} which provides information about the current running process. */
+public class ProcessResource extends ResourceProvider {
+  @Override
+  protected Attributes getAttributes() {
+    Attributes.Builder attributes = Attributes.newBuilder();
+
+    // TODO(anuraaga): Use reflection to get more stable values on Java 9+
+    RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+    long pid = -1;
+    String runtimeName = ManagementFactory.getRuntimeMXBean().getName();
+    int atIndex = runtimeName.indexOf('@');
```

Comment:
could you add a little comment about what the structure of this name looks like, so the code doesn't just look like magic. Thanks!